Source/Evolve/WarpXEvolve.cpp

```diff
@@ -401,20 +289,18 @@ WarpX::Evolve (int numsteps)
                       << " s; Avg. per step = " << evolve_time/(step-step_begin+1) << " s\n\n";
         }
 
-        exit_loop_due_to_interrupt_signal = SignalHandling::TestAndResetActionRequestFlag(SignalHandling::SIGNAL_REQUESTS_BREAK);
```

Comment:
I would keep this here since it makes it clearer what is happening (and keeps that flag local).

Reply:
I find the whole if condition super unreadable, moving it (and further growth how to stop a simulation) into a separate call helps sub-structure it, I find.
I do not think that signal handling logic belongs explicitly here, it is just one out of many ways to end a sim.

Reply:
Ok. I would suggest a comment here that `checkStopSimulation` has the side effect of setting `m_exit_loop_due_to_interrupt_signal`, otherwise the behavior might seem mysterious.
BTW, I had submitted a PR to clean this up, but it was never merged, PR #3881. That would simply this so that the extra routine may not be needed.